tests/e2e/default/reports/reports-subject.wdio-spec.js

```diff
@@ -58,6 +58,29 @@ describe('Reports Subject', () => {
     expect(openReportInfo.senderPhone).to.equal(senderPhone);
   };
 
+  const baseReport = (reportId, form, userPhone, fields) => {
+    return {
+      _id: reportId,
+      form: form,
+      type: 'data_record',
+      from: userPhone,
+      fields: fields,
+    };
+  };
+
+  const createSmsReport = async (reportId, form, userPhone, message, fields, errors = [], ) => {
```

Comment:
Minor, but this is quite a lot of parameters for a function.  (I am a bit surprised that Sonar did not flag this. Maybe it does not apply that rule for test code?  🤔)

There are a couple different ways to approach this functionality (e.g. builder pattern, multiple function calls, etc). In this case, I would say the most easy thing to do might be to just accept the `report` as a parameter along with `userPhone` and `message`.  Then you can populate `report.sms_message` in this function (using the provided data) and save the doc.  The rest of the `report` object would need to be built by the code that consumes this function (could use the `baseReport` function, but honestly I am not sure that function is providing much value....).  With these kinds of utility functions there is a balance between convenience vs hiding too much so that you cannot tell what all the params are actually being used for...

Reply:
Thank you for this feedback Josh! I will make the changes! 🫶